resources/pdns_authoritative_service_sysvinit.rb

```diff
@@ -0,0 +1,93 @@
+#
+# Cookbook Name:: pdns
+# Resources:: pdns_authoritative_service
+#
+# Copyright 2017, Aetrion, LLC DBA DNSimple
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+include ::Pdns::PdnsAuthoritativeHelpers
+
+resource_name :pdns_authoritative_service_sysvinit
+
+provides :pdns_authoritative_service, os: 'linux' do |_node|
+  Chef::Platform::ServiceHelpers.service_resource_providers.include?(:debian) ||
```

Comment:
Sorry but I don't understand what you want me to do? What is the resource DSL way? I am just using what chef is also using in certain [cookbooks](https://github.com/chef-cookbooks/httpd/blob/master/libraries/httpd_service_debian_systemd.rb#L8)